### PR TITLE
E2E Test set up / authentication for first admin

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

This test simulates the experience a user would have when setting up Site Kit for the first time:
* Navigating to the splash page
* Providing client configuration
* Signing in with Google (Google oAuth is bypassed)
* Returning to WP and landing on the Site Kit dashboard with a Congratulations message

Addresses issue #250

## Checklist

- [ ] My code is tested and passes existing unit tests. N/A
- [ ] My code has an appropriate set of unit tests which all pass. N/A
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by lucymtc** [comment]
@googlebot consent

**Comment by felixarntz** [comment]
@aaemnnosttv @lucymtc Just noticed this one randomly, fyi this Googlebot issue apparently happens every time more than one person is involved with a PR. If that ever is the case, ping me so I can resolve it. Just did it now.

**Comment by swissspidy** [comment]
@felixarntz They can also just leave a comment as suggested by Googlebot. Works like a charm 🙂

File: tests/e2e/specs/auth-flow.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		} );
+	} else {
+		request.continue();
+	}
+}
+
+describe( 'Site Kit set up flow for the first time', () => {
+
+	beforeAll( async() => {
+		await deactivateAllOtherPlugins();
```
Comment (swissspidy) [inline tests/e2e/specs/auth-flow.test.js]: Can you explain the rationale behind this? Why is this needed?

If you're running into issues otherwise we'd need verify whether this happens for other tests as well. If so, we could automate this.

File: tests/e2e/specs/auth-flow.test.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		await resetSiteKit();
+		await activatePlugin( 'e2e-tests-oauth-callback-plugin' );
+	} );
+
+	afterAll( async() => {
+		await deactivateAllOtherPlugins();
+		await resetSiteKit();
```
Comment (swissspidy) [inline tests/e2e/specs/auth-flow.test.js]: Resetting seems like a reasonable thing to do for every test. We could move this to the `bootstrap.js` file.

**Comment by swissspidy** [comment]
@aaemnnosttv Left two comments. If we can resolve these then I think this should be good to go.

File: tests/e2e/specs/auth-flow.test.js (hunk 1 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/auth-flow.test.js]: The main reason is that core's `deactivatePlugin` will fail if the plugin is already deactivated (unlike `activatePlugin`). This test uses multiple utilities so this is an idempotent way to ensure that they're all deactivated without needing to check if they're active or not. It makes more sense in the context of a local environment where we're running the same tests over and over again, and different plugins could be still active or not from the previous run but I think it's a reasonable utility to have.

File: tests/e2e/specs/auth-flow.test.js (hunk 2 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/auth-flow.test.js]: Yeah, I agree. Should this be done as part of this PR (and change other existing tests) or should we have a dedicated issue for this refactoring?

Comment (swissspidy) [reply tests/e2e/specs/auth-flow.test.js]: Let's add a separate issue for that. There seem to be a few possible ways to solve this, so it would be good to clarify in an implementation brief before diving right into it.

**Comment by lucymtc** [comment]
@googlebot I consent

File: tests/e2e/specs/auth-flow.test.js (hunk 1 above)
Comment (swissspidy) [reply tests/e2e/specs/auth-flow.test.js]: > The main reason is that core's `deactivatePlugin` will fail if the plugin is already deactivated

Sounds like something that should be reported upstream 🙂 

See https://github.com/WordPress/gutenberg/pull/16816

> It makes more sense in the context of a local environment where we're running the same tests over and over again, and different plugins could be still active or not from the previous run but I think it's a reasonable utility to have.

Another thing to put into `bootstrap.js`?

To clarify, you're not running the tests in the isolated Docker container?

Comment (aaemnnosttv) [reply tests/e2e/specs/auth-flow.test.js]: > Sounds like something that should be reported upstream 🙂

Thanks! 😄 

> Another thing to put into bootstrap.js?

Yeah, it seems like that would be useful as well. That way, each test would start knowing only Site Kit is active.

> To clarify, you're not running the tests in the isolated Docker container?

I am running them in the Docker container, but it doesn't fully reset the state on each run and it would be too inefficient to run `env:reset-site` every time.

File: tests/e2e/specs/auth-flow.test.js (hunk 2 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/auth-flow.test.js]: Added https://github.com/google/site-kit-wp/issues/353
